pkgs/by-name/ni/nixos-render-docs-redirects/src/nixos_render_docs_redirects/__init__.py

```diff
@@ -0,0 +1,134 @@
+        raise IdentifierExists(identifier)
+
+    # Insert the new identifier in alphabetical order
+    new_redirects = list(redirects.items())
+    insertion_index = 0
+    for i, (key, _) in enumerate(new_redirects):
+        if identifier > key:
+            insertion_index = i + 1
+        else:
+            break
+    new_redirects.insert(insertion_index, (identifier, [f"{path}#{identifier}"]))
+    return dict(new_redirects)
+
+
+def move_content(redirects: dict[str, list[str]], identifier: str, path: str) -> dict[str, list[str]]:
+    if identifier not in redirects:
+        raise IdentifierNotFound(identifier)
+    redirects[identifier].insert(0, f"{path}#{identifier}")
+    return redirects
+
+
+def rename_identifier(
+    redirects: dict[str, list[str]],
+    old_identifier: str,
+    new_identifier: str
+) -> dict[str, list[str]]:
+    if old_identifier not in redirects:
+        raise IdentifierNotFound(old_identifier)
+    if new_identifier in redirects:
+        raise IdentifierExists(new_identifier)
+
+    # To minimise the diff, we recreate the redirects mapping allowing
+    # the new key to be updated in-place, preserving the index.
+    new_redirects = {}
+    current_path = ""
+    for key, value in redirects.items():
+        if key == old_identifier:
+            new_redirects[new_identifier] = value
+            current_path = value[0].split('#')[0]
+            continue
+        new_redirects[key] = value
+    new_redirects[new_identifier].insert(0, f"{current_path}#{new_identifier}")
+    return new_redirects
+
+
+def remove_and_redirect(
+    redirects: dict[str, list[str]],
+    old_identifier: str,
+    new_path: str,
+    new_identifier: str
+) -> dict[str, list[str]]:
+    if old_identifier not in redirects:
+        raise IdentifierNotFound(old_identifier)
+    if new_identifier not in redirects:
+        raise IdentifierNotFound(new_identifier)
+    redirects[new_identifier].extend(redirects.pop(old_identifier))
+    return redirects
+
+
+def main():
+    parser = argparse.ArgumentParser(description="redirects manipulation for nixos manuals")
+    commands = parser.add_subparsers(dest="command", required=True)
+
+    add_content_cmd = commands.add_parser("add-content")
+    add_content_cmd.add_argument("identifier", type=str)
+    add_content_cmd.add_argument("path", type=str)
+
+    move_content_cmd = commands.add_parser("move-content")
+    move_content_cmd.add_argument("identifier", type=str)
+    move_content_cmd.add_argument("path", type=str)
+
+    rename_id_cmd = commands.add_parser("rename-identifier")
+    rename_id_cmd.add_argument("old_identifier", type=str)
+    rename_id_cmd.add_argument("new_identifier", type=str)
+
+    remove_redirect_cmd = commands.add_parser("remove-and-redirect")
+    remove_redirect_cmd.add_argument("identifier", type=str)
+    remove_redirect_cmd.add_argument("new_location", type=str)
+
+    args = parser.parse_args()
+
+    with open("@redirects_file@") as file:
```

Comment:
It would be great if the script worked standalone without patching. In principle this can be passed as a mandatory flag, and then in the shell we create a wrapper that specifies the argument to be the right thing.

Reply:
A wrapper using something like `writeShellScriptBin`?

Reply:
Yes, that would also obviate the package parameter.